src/models/Order.js

```diff
@@ -60,47 +60,29 @@ const orderSchema = new mongoose.Schema(
     // The paymentIntentId field stores the Stripe state of the purchase
     paymentIntentId: String, // Payment intent ID is a string
     orderItems: [orderItemSchema],
+    expiresAt: {
+      type: Date,
+      default: Date.now() + 60 * 1000, // Initial time for 1 minute
+      index: { expires: '1m' }, // Index for automatic deletion after 1 minute
+    },
```

Comment:
Oh interesting. What does this do / why did you remove the other code? It looks like this will just fully delete any order that's older than 1 minutes, regardless of status; I don't _think_ we want that?

Reply:
I've considered using TTL (Time To Live) to automatically remove orders from the database, and I'd like to propose the following approach:

Orders with the status 'pending' will automatically change to 'canceled' after 1 minute using setInterval.

For orders that are 'canceled' and are no longer needed, we will add an 'expiresAt' field. These canceled orders will be automatically removed from the database 2 minutes after their cancellation. This helps reduce the database load and keeps it in good shape.

This approach will allow us to efficiently manage orders and store only relevant data in the database. If this approach works for us, we can adjust the timing, but for testing purposes, I'll keep it at 1 minute and 2 minutes. If this approach isn't suitable, I can remove this code, and the rest of the code will remain unchanged.
<img width="560" alt="expires" src="https://github.com/Code-the-Dream-School/dd-prac-team4-back/assets/95896159/0b56bf80-f188-4cc1-b828-de5dd13e0f0b">

Reply:
Ah that's a great idea! 

  
I think my concern is what you have currently doesn't tell Mongo to only delete orders with the `"cancelled"` status as far as I can tell :thinking:

Reply:
Ah I did some reading. So my understanding is that it will only do the automatic deletion if that value is set; is that what you got too? 

So the idea is:

*   by default the order is created with value `pending`
*   if it's still `pending` after **1 minute** then we automatically set the value to `cancelled`
*   if after **2 minutes** past it's creation date, it's still around, it will be deleted automatically

What happens if:

*   order is created (status  = `pending`)
*   order is updated by UI to status = `payment_successful`
*   2 minutes pass. Won't it still get deleted ? I guess in step 2 we have to ensure that we set `expiresAt` to `null` right?

Reply:
<img width="639" alt="expiresAt" src="https://github.com/Code-the-Dream-School/dd-prac-team4-back/assets/95896159/cf5759ac-19c3-4fb4-8dcd-adadbb3841ba">
@akosasante code sets up a special rule in MongoDB, automatically deletes orders with a 'cancelled' status after 2 minutes.

orderSchema.index - starting to create index in the order schema.

{ expiresAt: 1 } - this index is for the 'expiresAt' field, and it should be sorted in ascending order.

{ expireAfterSeconds: 120 - time limit in seconds. If an order has a 'cancelled' status and it's been more than 2 minutes since it was created, MongoDB will delete it from the database.

partialFilterExpression: { orderStatus: 'cancelled' } - telling MongoDB that this rule only applies to documents where the 'orderStatus' is 'cancelled'.

So, this code ensures that only cancelled orders get automatically deleted from the database after 2 minutes from their creation. This helps keep the database clean and free up space from unnecessary data.